zenodo/modules/spam/templates/zenodo_spam/safelist/admin.html

```diff
@@ -0,0 +1,275 @@
+                              {{ url_for('zenodo_spam.delete',
+                              user_id=user.id)}}"
+                            >Delete</a>
+                        </td>
+                    </tr>
+                  {% endif %}
+              {% endfor %}
+          </tbody>
+          <tfoot>
+              <tr>
+                  <th></th>
+                  <th>ID</th>
+                  <th>Username</th>
+                  <th>Email</th>
+                  <th>Name</th>
+                  <th>Whitelist</th>
+                  <th>Delete</th>
+              </tr>
+        </tfoot>
+    </table>
+      </div>
+  </div>
+</div>
+{%- endblock page_body %}
+
+{%- block javascript %}
+<!-- POC -->
+
+<script src="https://code.jquery.com/jquery-3.5.1.js"></script>
+<script src="https://cdn.datatables.net/1.12.1/js/jquery.dataTables.min.js"></script>
+<script src="https://cdn.datatables.net/select/1.4.0/js/dataTables.select.min.js"></script>
+<script src="https://cdn.datatables.net/1.12.1/js/dataTables.bootstrap.min.js"></script>
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+<link rel="stylesheet" href="https://cdn.datatables.net/1.12.1/css/jquery.dataTables.min.css">
+<link rel="stylesheet" href=https://cdn.datatables.net/select/1.4.0/css/select.dataTables.min.css">
+<link rel="stylesheet" href="https://cdn.datatables.net/1.12.1/css/dataTables.bootstrap.min.css">
+
+<script>
+  $(document).ready(function () {
+    var external_table = $('#external_table').DataTable({
+      columnDefs: [{
+        orderable: false,
+        className: 'select-checkbox',
+        targets: 0
+      }],
+      select: {
+        style: 'os',
+        selector: 'td:first-child'
+      },
+      order: [[1, 'asc']]
+    });
+
+    var no_external_table = $('#no_external_table').DataTable({
+      columnDefs: [{
+        orderable: false,
+        className: 'select-checkbox',
+        targets: 0
+      }],
+      select: {
+        style: 'os',
+        selector: 'td:first-child'
+      },
+      order: [[1, 'asc']]
+    });
+
+    $('#delete_bulk_external').click(function () {
+      {# TODO How to access the multi-selected values
+      https://datatables.net/extensions/select/integration #}
+    alert('Delete users')
+    })
+    $('#safelist_bulk_external').click(function () {
+      {# Suggestion: make safelist_add_remove() also accept list of ids and
+      run according to type #}
+      alert('Safelist users')
+    })
+    $('#delete_bulk_no_external').click(function () {
+      const data = no_external_table.rows({selected: true}).data().toArray();
+      if (data.length === 0) {
+        console.log('No items selected')
+      }
+
+      let userIds = []
+      data.forEach(row => {
+        userIds.push(row[1])
+      })
+
+      $.post( "{{ url_for('zenodo_spam.safelist_bulk_add') }}",
+        {'user_ids': userIds, 'next': {{ url_for('zenodo_spam.safelist_admin')
+        }} })
```

Comment:
```suggestion
        {'user_ids': userIds})
```

`next` and redirects are not that necessary here, since we're making an AJAX call.

Reply:
Once again, WIP from the copy pasting, indeed we can handle it on the frontend